api-admin/src/test/java/bio/terra/pearl/api/admin/service/forms/SurveyExtServiceTests.java

```diff
@@ -1,17 +1,100 @@
+import java.util.Optional;
+import java.util.UUID;
 import org.junit.jupiter.api.Assertions;
 import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.web.servlet.MockMvc;
 
+@ContextConfiguration(classes = SurveyExtService.class)
+@WebMvcTest
 public class SurveyExtServiceTests {
-  private SurveyExtService emptyService = new SurveyExtService(null, null, null);
+
+  @Autowired private MockMvc mockMvc;
+  @Autowired private SurveyExtService surveyExtService;
+
+  @MockBean private AuthUtilService mockAuthUtilService;
+  @MockBean private SurveyService mockSurveyService;
+  @MockBean private StudyEnvironmentSurveyService studyEnvironmentSurveyService;
 
   @Test
   public void createNewVersionRequiresSuperuser() {
     AdminUser user = AdminUser.builder().superuser(false).build();
     Assertions.assertThrows(
-        PermissionDeniedException.class, () -> emptyService.createNewVersion("foo", null, user));
+        PermissionDeniedException.class,
+        () -> surveyExtService.createNewVersion("foo", null, user));
+  }
+
+  @Test
+  public void createRequiresSuperuser() {
+    AdminUser user = AdminUser.builder().superuser(false).build();
+    Assertions.assertThrows(
+        PermissionDeniedException.class, () -> surveyExtService.create("foo", null, user));
+  }
+
+  @Test
+  public void getRequiresPortalAuth() {
+    AdminUser user = AdminUser.builder().superuser(false).build();
+    when(mockAuthUtilService.authUserToPortal(user, "foo"))
+        .thenThrow(new PermissionDeniedException("test1"));
+    Assertions.assertThrows(
+        PermissionDeniedException.class, () -> surveyExtService.get("foo", "blah", 1, user));
+  }
+
+  @Test
+  public void getRequiresSurveyMatchedToPortal() {
+    AdminUser user = AdminUser.builder().superuser(false).build();
+    Portal portal = Portal.builder().shortcode("testSurveyGet").id(UUID.randomUUID()).build();
+    when(mockAuthUtilService.authUserToPortal(user, portal.getShortcode())).thenReturn(portal);
+    Survey matchedSurvey =
+        Survey.builder()
+            .stableId("testMatchedToPortal")
+            .version(1)
+            .portalId(portal.getId())
+            .build();
+    when(mockSurveyService.findByStableId(matchedSurvey.getStableId(), matchedSurvey.getVersion()))
+        .thenReturn(Optional.of(matchedSurvey));
+    Survey unmatchedSurvey =
+        Survey.builder()
+            .stableId("testMatchedToPortalUnmatched")
+            .version(1)
+            .portalId(UUID.randomUUID())
+            .build();
+    when(mockSurveyService.findByStableId(
+            unmatchedSurvey.getStableId(), unmatchedSurvey.getVersion()))
+        .thenReturn(Optional.of(unmatchedSurvey));
```

Comment:
Suggestion: I've had good experiences extracting functions for setting up mocks like this. The creation and setup of `matchedSurvey` and `unmatchedSurvey` are nearly identical, and the volume of code here detracts from the clarity if the test. This could be two lines of code:
```
    Survey matchedSurvey = configureMockSurvey("testMatchedToPortal", portal.getId());
    Survey unmatchedSurvey = configureMockSurvey("testMatchedToPortalUnmatched", UUID.randomUUID());
```
with the addition of this function:
```
  private Survey configureMockSurvey(String surveyStableId, UUID portalId) {
    Survey survey = Survey.builder()
            .stableId(surveyStableId)
            .version(1)
            .portalId(portalId)
            .build();
    when(mockSurveyService.findByStableId(survey.getStableId(), survey.getVersion()))
        .thenReturn(Optional.of(survey));
    return survey;
  }
```